salt/auth/__init__.py

```diff
@@ -405,6 +410,70 @@ def get_auth_list(self, load):
 
         return auth_list
 
+    def check_authentication(self, load, auth_type, key=None, show_username=False):
```

Comment:
You know, I really don't like these sorts of all-in-one functions that much. I'm wondering if it would be better to the callers of this and have them make the determination to call `authenticate_token` vs. `authenticate_eauth`, etc and then just get rid of this altogether. Thoughts?

Reply:
This is a solid point, but if they are separated we would want them to be right next to each other in the code sicne we do not want the basic logic to diverge.
I am leaning more toward keeping the function like it is from the perspective of keeping the same logic in one place to stave off each auth type from becoming different.

Reply:
ABC perhaps? Your point about divergence is sound but this is going to be tricky to maintain IMHO.

Reply:
Blech, I think we would get more risk here with an ABC approach.
All in all I think this is fine as is, and I think that we are splitting hairs, this code has hardly been touched in master.py for years anyway

Reply:
To give some context, I wrote this function this way to handle the _many_ places we're basically just checking for the type of authentication, authenticating based on that type, handling the error message, and then obtaining the correct "auth_list".

The `runner` and `wheel` functions in both `master.py` and `masterapi.py` have variations on this same theme. Which means changes to one issue has to be updated in at least 4 places (see #43283 for an example) instead of just one (line 453 in this PR). There are a couple more places that can use this function that I didn't include here yet because I wanted to get feedback on this first.

However, I see how this could be tricky to maintain moving forward. I can break this up into different functions if desired.

Reply:
I am ok with this as is and I think we should push any such changes to a new PR, also because the current state is most like how it has traditionally looked and future changes should likely be made in a new PR

Reply:
@thatch45 Yes, I agree the new changes should be in a new PR.

@cachedout Sound good to you?

Reply:
Yes, this is fine by me.